python/ray/_private/usage/usage_lib.py

```diff
@@ -124,3 +169,80 @@ def get_cluster_metadata(gcs_client, num_retries) -> dict:
             num_retries=num_retries,
         )
     )
+
+
+def generate_report_data(cluster_metadata: dict) -> dict:
+    """Generate the report data.
+
+    cluster_metadata (dict): The cluster metadata of the system generated by
+        `_generate_cluster_metadata`.
+
+    Returns:
+        UsageStats in a dictionary type.
+    """
+    data = UsageStats(
+        ray_version=cluster_metadata["ray_version"],
+        python_version=cluster_metadata["python_version"],
+        schema_version=cluster_metadata["schema_version"],
+        source=cluster_metadata["source"],
+        session_id=cluster_metadata["session_id"],
+        git_commit=cluster_metadata["git_commit"],
+        os=cluster_metadata["os"],
+        collect_timestamp_ms=int(time.time() * 1000),
+    )
+    return data
+
+
+class UsageReportClient:
+    """The client implementation for usage report.
+
+    It is in charge of writing usage stats to the directory
+    and report usage stats.
+    """
+
+    def _write_usage_data(self, data: UsageStats, dir_path: str) -> None:
+        """Write the usage data to the directory.
+
+        Params:
+            data (dict): Data to report
+            dir_path (Path): The path to the directory to write usage data.
+        """
+        assert _usage_stats_enabled()
```

Comment:
The env var could get changed dynamically, causing this assert to fire. It would be safer to remove this or return.

Reply:
Will return this!